### DISCUSSION TITLE
Doubt about contribution

### DISCUSSION THREAD

**Discussion by 11-aryan** [TheAlgorithms/Go#664] [General]
The description in readme file for segmenttree seems to be missing. I wanted to contribute to it. How can I do it? Should I create an issue first or directly create a PR ?

**Comment by raklaptudirm** [comment]
The description in the readme is auto-generated from package documentation. You need to add the relevant documentation for it to appear in the README.

**Comment by 11-aryan** [comment]
@raklaptudirm Sorry, I didn't quite catch that. Could you please elaborate?

**Reply by raklaptudirm** [reply]
In Go, you write package documentation as [comments](https://go.dev/doc/comment). We use an automatic tool which populates the readme with this documentation. The package lacks proper documentation if the documentation is missing from the README. Therefore, you will have to add the missing documentation to the package and it will be automatically added to the README.

**Reply by 11-aryan** [reply]
ok got it! Thanks
